workflow-core/src/commonMain/kotlin/com/squareup/workflow1/BaseRenderContext.kt

```diff
@@ -173,6 +173,31 @@ public interface BaseRenderContext<out PropsT, StateT, in OutputT> {
    * as a debugging aid
    * @param update Function that defines the workflow update.
    */
+  @Suppress("DEPRECATION")
+  public fun eventHandler(
+    name: String,
+    // Type variance issue: https://github.com/square/workflow-kotlin/issues/891
+    update: WorkflowAction<
+      @UnsafeVariance PropsT,
+      StateT,
+      @UnsafeVariance OutputT
+      >.Updater.() -> Unit
+  ): () -> Unit = eventHandler({ name }, update)
```

Comment:
I'm confused why this one delegates to the original `name: () -> String = { "eventHandler" }` one that we have deprecated? 

Why isn't this one like this:

```suggestion
  ): () -> Unit {
    return {
      actionSink.send(action(name, update))
    }
  }
```

Reply:
D'oh.

Reply:
Oh, because that would make it recursive. I'd have to make this PR even bigger.

Thinking about it, _why_ deprecate the lazy ones? The hard part was getting all these new entry points defined; can definitely see a benefit to having the lazy option; and you'll note I didn't deprecate the fundamental lazy `action {}` variants.

I guess I was…being lazy.

Reply:
I'm ok with keeping the lazy ones, but we can't keep them as is because they have the default `= { "eventHandler" }` which is what we're trying to avoid, no?

Reply:
Ah of course. Will fix.

Reply:
Bah. I don't have lazy ones for `safeEventHandler` because it seemed ridiculous. 

Flip flopping again. Let's deprecate them, and I'll fix the deprecation message.

The 0.01% of places where laziness might possibly matter can just use `action({}) {}` directly.